tree/ntuple/v7/test/ntuple_compat.cxx

```diff
@@ -58,3 +64,86 @@ TEST(RNTupleCompat, FeatureFlag)
       EXPECT_THAT(err.what(), testing::HasSubstr("unsupported format feature: 137"));
    }
 }
+
+TEST(RNTupleCompat, FwdCompat_FutureNTuple)
+{
+   using ROOT::Experimental::RXTuple;
+
+   constexpr static const char *kNtupleObjName = "ntpl";
+
+   FileRaii fileGuard("test_ntuple_compat_fwd_compat_future_ntuple.root");
+
+   // Write an RXTuple to disk. It is a simulacrum of a future version of RNTuple, with additional fields and a higher
+   // class version.
+   {
+      auto file = std::unique_ptr<TFile>(
+         TFile::Open(fileGuard.GetPath().c_str(), "RECREATE", "", ROOT::RCompressionSetting::ELevel::kUncompressed));
+
+      auto xtuple = RXTuple{};
+      auto key = TKey(&xtuple, RXTuple::Class(), kNtupleObjName, sizeof(RXTuple), file.get());
+      key.WriteFile();
+      file->Close();
+   }
+
+   // Patch all instances of 'RXTuple' -> 'RNTuple'.
+   // We do this by just scanning the whole file and replacing all occurrences.
+   // This is not the optimal way to go about it, but since the file is small (~1KB)
+   // it is fast enough to not matter.
+   {
+      FILE *f = fopen(fileGuard.GetPath().c_str(), "r+b");
+
+      fseek(f, 0, SEEK_END);
+      std::size_t fsize = ftell(f);
+
+      char *filebuf = new char[fsize];
```

Comment:
Let's use a unique pointer

Reply:
I find that in this case it comes with more downsides than upsides, since it increases the code noise quite a bit (all the `.get()` and similar) and it only saves 1 line of code, which is basically impossible to get wrong anyway since the lifetime is so short

Reply:
Indeed correct in this case.  In the more general case, using `new/delete` explicitly in not exception safe.  (To avoid the get you can consider caching the value in a variable).

Reply:
Ok, fine by me then